BinaryEncoding.md

```diff
@@ -3,120 +3,143 @@
-  * Instead, a module defines its *own* local index spaces of operators by providing tables *of names*. 
-    * So what the spec *would* define is a set of names and their associated semantics.
-  * To avoid (over time) large index-space declaration sections that are largely the same
-    between modules, finalized versions of standards would define named baseline index spaces
-    that modules could optionally use as a starting point to further refine.
-    * For example, to use all of [the MVP](MVP.md) plus
-      [SIMD](PostMVP.md#fixed-width-simd) the declaration could be "base"
-      followed by the list of SIMD operators used.
-    * This feature would also be most useful for people handwriting the [text format](TextFormat.md).
-    * However, such a version declaration does not establish a global "version" for the module
-      or affect anything outside of the initialization of the index spaces; decoders would
-      remain versionless and simply add cases for new *names* (as with current JavaScript parsers).
-
-## Proposals
-
-The native prototype built for [V8](https://github.com/WebAssembly/v8-native-prototype)
-implements a binary format that embodies most, but not all of the ideas in this document.
-It is described in detail in a [public design doc](https://docs.google.com/a/google.com/document/d/1761v1AfhFM5kE8NArF_PyXcl-iVh0Dx3InOrmcyIoiI/pub) and a [copy of the original](https://docs.google.com/document/d/1-G11CnMA0My20KI9D7dBR6ZCPOBCRD0oCH6SHCPFGx0/edit?usp=sharing).
+### Stream splitting
+Refers to splitting the single encoded binary stream out into smaller streams, partitioned based on element type or semantic information.
+Research has shown that splitting constants, names, and opcodes into their own streams increases the effectiveness of generic compression.
+
+### Subtree deduplication / nullary macros
+Identifies and prunes structurally identical nodes and trees of nodes. Most applications contain significant amounts of structural 
+duplication that is not completely erased by generic compression.
+**Non-nullary macros** are an extension of this technique that enables further compression at the cost of additional complexity.
+
+### Index tables
+Modules contain multiple index tables that assign indexes to key pieces of information like opcodes or data types. This enables
+compatibility between implementations and allows information to be represented more efficiently.
+
+### Sections
+Modules are split up into sections with well-defined contents that can refer to each other and are identified by name.
+The use of names allows new section types to be introduced in the future.
+
+### Strings
+Strings are encoded as null-terminated [UTF8](http://unicode.org/faq/utf_bom.html#UTF8).
+
+# v8-native module structure
+
+The following documents the current v8-native prototype format, not the binary encoding intended for standardization.
+
+## High-level structure
+A module contains (in this order):
```

Comment:
You can just merge this sub point into the main point. "A module is sequence of sections, each identified by a unique section type identifier".

Please also specify the constants we are using so far, in the interest of making the document detailed enough to implement a decoder.